src/pages/api/comments.ts

```diff
@@ -0,0 +1,151 @@
+import { getSession } from "@auth0/nextjs-auth0";
+import { Prisma } from "@prisma/client";
+import { withSentry } from "@sentry/nextjs";
+import { NextApiHandler } from "next";
+
+import { prisma } from "../../shared/prisma-helper";
+import { generateAvatarUrl } from "../../shared/user-helpers";
+import {
+  CommentsApiHandlerSuccessfulGetResponseBody,
+  CommentsApiHandlerSuccessfulPostResponseBody,
+  NewComment,
+  PublicCommentInfo,
+} from "../../types";
+
+const commentSelect = {
+  accidentId: true,
+  author: {
+    select: {
+      name: true,
+      email: true,
+      picture: true,
+    },
+  },
+  createdAt: true,
+  id: true,
+  isPublished: true,
+  text: true,
+} as const;
+
+const convertSelectedCommentToPublicInfo = ({
```

Comment:
We select a joined version of Comment + User from Prisma. To avoid complexity on the client, we flatten the result into a single objects containing only scalar fields.